src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/totrinnskontroll/TotrinnskontrollService.kt

```diff
@@ -1,61 +1,102 @@
 import no.nav.tilleggsstonader.sak.infrastruktur.exception.Feil
 import no.nav.tilleggsstonader.sak.infrastruktur.sikkerhet.BehandlerRolle
 import no.nav.tilleggsstonader.sak.infrastruktur.sikkerhet.SikkerhetContext
 import no.nav.tilleggsstonader.sak.infrastruktur.sikkerhet.SikkerhetContext.NAVIDENT_REGEX
 import no.nav.tilleggsstonader.sak.tilgang.TilgangService
+import no.nav.tilleggsstonader.sak.vedtak.totrinnskontroll.domain.TotrinnInternStatus
+import no.nav.tilleggsstonader.sak.vedtak.totrinnskontroll.domain.Totrinnskontroll
+import no.nav.tilleggsstonader.sak.vedtak.totrinnskontroll.domain.TotrinnskontrollRepository
+import no.nav.tilleggsstonader.sak.vedtak.totrinnskontroll.domain.Årsaker
 import no.nav.tilleggsstonader.sak.vedtak.totrinnskontroll.dto.BeslutteVedtakDto
 import no.nav.tilleggsstonader.sak.vedtak.totrinnskontroll.dto.StatusTotrinnskontrollDto
 import no.nav.tilleggsstonader.sak.vedtak.totrinnskontroll.dto.TotrinnkontrollStatus
 import no.nav.tilleggsstonader.sak.vedtak.totrinnskontroll.dto.TotrinnskontrollDto
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
-import java.util.*
+import java.util.UUID
 
 @Service
 class TotrinnskontrollService(
     private val behandlingshistorikkService: BehandlingshistorikkService,
     private val behandlingService: BehandlingService,
     private val tilgangService: TilgangService,
+    private val totrinnskontrollRepository: TotrinnskontrollRepository,
 ) {
+    @Transactional
+    fun sendtilBeslutterSteg(saksbehandling: Saksbehandling) {
+        val eksisterandeTotrinnskontroll = totrinnskontrollRepository.findTopByBehandlingIdOrderBySporbarEndretEndretTidDesc(saksbehandling.id)
+        if (eksisterandeTotrinnskontroll == null ||
+            eksisterandeTotrinnskontroll.status == TotrinnInternStatus.ANGRET ||
+            eksisterandeTotrinnskontroll.status == TotrinnInternStatus.UNDERKJENT
+        ) {
+            val nyTotrinnskontroll = Totrinnskontroll(
+                behandlingId = saksbehandling.id,
+                sporbar = Sporbar(),
+                saksbehandler = SikkerhetContext.hentSaksbehandlerEllerSystembruker(),
+                status = TotrinnInternStatus.KAN_FATTE_VEDTAK,
+            )
+            totrinnskontrollRepository.insert(nyTotrinnskontroll)
+        } else {
+            throw Feil(
+                message = "Totrinnskontroll har feil status for behandling status = ${eksisterandeTotrinnskontroll.status}",
+                frontendFeilmelding = "Får ikke sendt oppgave til totrinnskontroll",
+            )
+        }
+    }
+
+    @Transactional
+    fun angreSendTilBeslutter(behandlingId: UUID) {
+        val eksisterandeTotrinnskontroll = totrinnskontrollRepository.findTopByBehandlingIdOrderBySporbarEndretEndretTidDesc(behandlingId)
+        if (eksisterandeTotrinnskontroll?.status == TotrinnInternStatus.KAN_FATTE_VEDTAK) {
```

Comment:
Her går det å vurdere å bruke `feilHvis`, som kun tar en feilmelding, tenker vi ikke trenger 2 ulike feilmeldinger for hva som logges og hva som vises til saksbehandleren